### PR TITLE
Activate Pixel Charge Reweighting for phase-0 detector (2016 only)

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

In the scope of the preparations for the 2016 Ultra-Legacy MC campaigns, when looking at the data charge profiles, we have reached the conclusion that the radiation damage accumulated by the Phase-0 detector justifies the activation of the pixel cluster charge re-weighting (full details on the study of the radiation damage is presented here: [here](https://indico.cern.ch/event/808418/contributions/3495567/attachments/1878023/3093308/ultra_legacy_templates_2016_Update.pdf)).
This feature was up to now nominally switched off for the the Phase-0 detector and it is hereby activated by means of the era mechanism. 
A new modifier `Configuration/Eras/python/Modifier_pixel_2016_cff.py ` is introduced and used in the definition of the `Era_Run2_2016` to selectively switch on the feature only for 2016 which is targeted by the Ultra-Legacy.
An appropriate set of 2D template conditions for the radiation damage model has been derived together with update CPE conditions (Simulation Lorentz Angle, Lorentz Angle offset calibration, empry Lorentz Angle widths, generic errors and 1D templates), using the third IOV of the data conditions as an approximation of the "average" radiation damage in 2016.
They are proposed in this PR, by updating several `autoCond` keys with candidate Global Tags:
   * `run2_design`:  https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_design_v1/110X_mcRun2_design_v2
   * `run2_mc`: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_asymptotic_v1/110X_mcRun2_asymptotic_v2
   * `run2_mc_cosmics`: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2cosmics_startup_deco_v1/110X_mcRun2cosmics_startup_deco_v2
   * `run2_mc_pa`: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_pA_v1/110X_mcRun2_pA_v2

**N.B.**: in the design MC case only the 2D template tags are changed using the same payload both at the `numerator` and `denominator` label, resulting in no re-weigthing applied (design is assumed to be unirradiated).
~~If AlCa/DB can make these full-fledged Global Tags I will be gladly updating the autoCond file~~.

#### PR validation:
The PR passes standard tests (`addOnTests` and ```runTheMatrix.py -l limited -t 4 -j 8```.

Additionally the effect of the charge re-weighting has been tested in simulation by comparing the pixel charge profiles vs depth with two 2016 runs 278240 & 278308 ( falling into IOV 3  of the data CPE conditions) and the reference MC without re-weighting.
The comparison is shown [here](https://indico.cern.ch/event/808419/contributions/3509501/attachments/1884963/3106810/ccp_2016ULprep_reweighting_newTemplate.pdf)

image

Post-reweighting DATA and MC profiles are found in satisfactory agreement. 

#### if this PR is a backport please specify the original PR:

This PR is not a backport
cc:
@tsusa @tvami @leaca

### CODE REVIEW

**Comment by christopheralanwest** [comment]
please test

**Comment by civanch** [comment]
please test

File: Configuration/Eras/python/Era_Run2_2016_cff.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 from Configuration.Eras.Modifier_run2_muon_2016_cff import run2_muon_2016
 from Configuration.Eras.Modifier_run2_egamma_2016_cff import run2_egamma_2016
 from Configuration.Eras.Modifier_run2_L1prefiring_cff import run2_L1prefiring
+from Configuration.Eras.Modifier_pixel_2016_cff import pixel_2016
 
 Run2_2016 = cms.ModifierChain(run2_common, run2_25ns_specific,
- stage2L1Trigger, ctpps_2016, run2_HLTconditions_2016, run2_muon_2016, run2_egamma_2016, run2_L1prefiring)
+ stage2L1Trigger, ctpps_2016, run2_HLTconditions_2016, run2_muon_2016, run2_egamma_2016, run2_L1prefiring, pixel_2016)
```
Comment (makortel) [inline Configuration/Eras/python/Era_Run2_2016_cff.py]: I suppose `pixel_2016` should be removed from `Run2_2017`?
https://github.com/cms-sw/cmssw/blob/72e9ddf93ef02082d4c7abc4569fb3b43ea2ea7b/Configuration/Eras/python/Era_Run2_2017_cff.py#L21

(on the other hand, with this many year-specific modifiers maybe we should consider reorganizing the era definitions by separating the modifiers that carry on to the next year and the modifiers that are specific for that year; clearly not for this PR)

Comment (mmusich) [reply Configuration/Eras/python/Era_Run2_2016_cff.py]: > I suppose pixel_2016 should be removed from Run2_2017?

why?
anyway it is already overwritten by:
https://github.com/cms-sw/cmssw/blob/72e9ddf93ef02082d4c7abc4569fb3b43ea2ea7b/SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py#L112

Comment (makortel) [reply Configuration/Eras/python/Era_Run2_2016_cff.py]: Easier maintenance. If the `pixel_2016` gets used more, one doesn't have to undo the modifications with `phase1Pixel`.

In general my understanding for the modifiers with the year postfix is that they are specific for that year and are not intended to be enabled for subsequent years (but I could be wrong, but then I'd say we have a mess).

Comment (mmusich) [reply Configuration/Eras/python/Era_Run2_2016_cff.py]: OK, thanks. I was actually asking myself if undoing it was the recommend way when preparing the PR, but then decided for economy not to do it. If you confirm that's the desired outcome I will update the branch. I need also updated GTs from @christopheralanwest. I will push all the updates at the same time.

**Comment by christopheralanwest** [comment]
I've created GTs based on the candidates proposed here, plus an additional tag `SiPixelLorentzAngle_forWidth_phase0_empty_mc` with label `forWidth`, as discussed at the 29 July AlCa/DB meeting [1]:

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_design_v2/110X_mcRun2_design_Candidate_2019_07_24_16_36_56 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_asymptotic_v2/110X_mcRun2_asymptotic_Candidate_2019_07_24_12_20_37 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2cosmics_startup_deco_v2/110X_mcRun2cosmics_startup_deco_Candidate_2019_07_24_12_21_07 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/110X_mcRun2_pA_v2/110X_mcRun2_pA_Candidate_2019_07_24_12_21_09 

Please use these GTs in your autoCond update.

[1] https://indico.cern.ch/event/835589/contributions/3513525/attachments/1887574/3112193/20190729_AlCaDB_UL16.pdf

File: Configuration/Eras/python/Era_Run2_2016_cff.py (hunk 1 above)
Comment (mmusich) [reply Configuration/Eras/python/Era_Run2_2016_cff.py]: comment has been taken care of by https://github.com/cms-sw/cmssw/pull/27600/commits/1bc910f8fee7055135bcd89f32211c81699b4a7f

**Comment by mmusich** [comment]
> I've created GTs based on the candidates proposed here, plus an additional tag `SiPixelLorentzAngle_forWidth_phase0_empty_mc` with label `forWidth`, as discussed at the 29 July AlCa/DB meeting

@christopheralanwest thanks.
GTs are updated by https://github.com/cms-sw/cmssw/pull/27600/commits/7de1d31a1145d7b81b1666b2cefad962db93f5db. PR description is updated accordingly.

**Comment by christopheralanwest** [comment]
please test

**Comment by christopheralanwest** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by kpedro88** [comment]
merge

**Comment by fabiocos** [comment]
+operations

the update is coherent with the purpose of the PR

**Pull request merged**